blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/QueueContainerImpl.java

```diff
@@ -33,9 +39,39 @@ public BlueQueueItem get(String name) {
                 return blueQueueItem;
             }
         }
+        // JENKINS-38540 - To make this consistent with the activity API, check the runs, too
+        String runId = findQueueInRuns(name);
+        if (runId != null) {
+            try {
+                StaplerResponse rsp = Stapler.getCurrentResponse();
+                // Send a redirect, not sure the specific code which would be best. substring to fix double slash
+                rsp.sendRedirect(Links.ensureTrailingSlash(Jenkins.getInstance().getRootUrl() + pipeline.getLink().toString().substring(1) + "runs/" + runId));
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            return null;
+        }
         return null;
     }
 
+    /**
+     * Finds a queue in the runs based on the queueId
+     */
+    private String findQueueInRuns(String name) {
```

Comment:
`findQueueInRuns` ? Maybe `findRunIdByQueueId` might more accurately describe what this is doing?